src/reports/library/records.cpp

```diff
@@ -10,18 +10,37 @@ int records_clear() {
     return 0;
 }
 
-int records_add_report(const char* reportName, uint64_t node_id, uint64_t gid, uint64_t vgid,
+int records_add_report(const char* reportName, uint64_t node_id, uint64_t /* gid */, uint64_t /* vgid */,
                        double tstart, double tend, double dt, int mapping_size, const char* kind,
                        int attributes_mapping_size, char* unit) {
     // if we are adding reports, then we should reset the flag for sharing reports
     ReportingLib::first_report = true;
 
-    return InitReports.add_report(std::string(reportName), node_id, gid, vgid, tstart, tend, dt, std::string(kind));
+    try {
+        if (!InitReports.report_exists(reportName)) {
+            InitReports.create_report(reportName, kind, tstart, tend, dt);
+        }
+        auto report = InitReports.get_report(reportName);
+        report->add_node(node_id);
+    } catch (const std::exception& err) {
+        logger->error(err.what());
+    }
+
+    return 0;
 }
 
 int records_add_var_with_mapping(const char* report_name, uint64_t node_id, double* voltage, int mapping_size, int* mapping_value) {
+    if (!InitReports.report_exists(report_name)) {
+        return 0;
+    }
+
     uint32_t element_id = mapping_value[0];
-    return InitReports.add_variable(std::string(report_name), node_id, voltage, element_id);
+    try {
+        InitReports.get_report(report_name)->add_variable(node_id, voltage, element_id);
```

Comment:
Maybe separate it in 2 lines like you did in add_report?
```suggestion
        auto report = InitReports.get_report(report_name);
        report->add_variable(node_id, voltage, element_id);
```